UnifiedSearch: Introduce a ResourceIndex interface and bleve implementation

ryantxu opened the pull request:
This extends work in https://github.com/grafana/grafana/pull/96281 -- and is part of the fully working [exploration](https://github.com/grafana/grafana/compare/main...unified-search-exploration) trying to detangle the existing startup + index management logic from the core resource server.

-----

This PR introduces an index interface that can be managed by the backend server without knowing how it is implemented.  This moves our existing bleve logic to a standalone implementation that can be tested independently.

Given that the protobuf definitions we have do not yet support the requests/response we need; this updates them to better fit the range of questions we will ask it

ryantxu committed:
add the bleve index

ryantxu committed:
Merge remote-tracking branch 'origin/unified-split-document-builder' into unified-split-bleve-index

ryantxu committed:
federated search

ryantxu committed:
with sort... but not working

ryantxu committed:
with sort... but not working

ryantxu committed:
sorting

ryantxu committed:
lint fix

ryantxu commented on pkg/storage/unified/search/bleve_test.go:
```diff
@@ -0,0 +1,281 @@
+				Key: key,
+			},
+			Limit: 100000,
+			// ????? this does not seem to work
+			SortBy: []*resource.ResourceSearchRequest_Sort{
+				{Field: "title", Desc: false}, // ccc,bbb,aaa
+			},
```
🤯 🤯 🤯 

this does not seem to be working for all the things i have tried

ryantxu committed:
move query setup out of main search function

ryantxu committed:
add deprecation notice

ryantxu committed:
add deprecation notice

ryantxu committed:
Merge remote-tracking branch 'origin/unified-split-document-builder' into unified-split-bleve-index

ryantxu committed:
merge upstream

renovate[bot] committed:
Update dependency eslint to v9.14.0 (#96382)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency eslint-plugin-jest to v28.9.0 (#96383)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

vtorosyan committed:
Docs: Clean up basic auth, anonymous access and main pages for auth configuration (#96259)

* Docs: Clean up basic auth, anonymous access and main pages for auth configuration

* Fix the typo

* Prettier should fix this

* fix minor typo

* Update docs/sources/setup-grafana/configure-security/configure-authentication/_index.md

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

---------

Co-authored-by: Irene Rodríguez <irene.rodriguez@grafana.com>
Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

renovate[bot] committed:
Update dependency eslint-plugin-jsdoc to v50.5.0 (#96385)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency eslint-plugin-no-barrel-files to v1.2.0 (#96387)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mdvictor committed:
Fix dashboards listing table appearing distorted (#96371)

* fix CSS

* approx height calc to account for header div

bergquist committed:
Loki: Adds a `suggestions` resource call that inject scopes for finding label/labelvalues (#96025)

Signed-off-by: bergquist <carl.bergquist@gmail.com>
Co-authored-by: Sven Grossmann <sven.grossmann@grafana.com>

MissingRoberto committed:
[Cloud Migrations] Fix panic in unit tests (#96389)

* Fix panic in unit test

* Reset mock to new one between tests

imatwawana committed:
Docs: Add auth entries to what's new 11.2 (#96353)

santihernandezc committed:
Alerting: Skip sanitizing labels when sending alerts to the remote Alertmanager (#96251)

* Alerting: Skip sanitizing labels when sending alerts to the remote Alertmanager

* fix drone image name

tomratcliffe committed:
Alerting: Fix some templates RBAC UI action control/checks (#94898)

leeoniya committed:
Matchers: Require explicit regexp syntax in byRefId matcher (#96358)

bohandley committed:
Revert "Prometheus: Check both error and http status for source of error " (#96396)

tolzhabayev committed:
Docs: fixing angular migration link (#96431)

fixing docs link

kalleep committed:
Zanana: Fix list test (#96401)

Fix: List operation are not guaranteed to respond in a certain order

renovate[bot] committed:
Update dependency i18next to v23.16.5 (#96392)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency eslint-plugin-testing-library to v6.4.0 (#96391)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

mgyongyosi committed:
Docs: Update GitLab, GitHub, Okta, Generic OAuth docs (#96386)

* GitHub docs changes

* Bunch of updates

* More updates

tomratcliffe committed:
Alerting: Fix permissions for timeintervals UI and improve display of contact points in notification policies (#96129)

Co-authored-by: Gilles De Mey <gilles.de.mey@gmail.com>

tskarhed committed:
Combobox: Documentation (#96307)

* Initial Combobobox docs

* Add test docs

* Add docs to props

* Add ComboboxOption docs

* Apply suggestions from code review

Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

---------

Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

brendamuir committed:
Alerting docs: adds cloud alert rule limits (#96441)

renovate[bot] committed:
Update dependency knip to v5.37.0 (#96435)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency msw to v2.6.4 (#96436)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

imatwawana committed:
Docs: update docs for singleTopNav (#95663)

* updated kiosk behaviour

* Removed TV mode and updated Normal mode

* Updated kiosk mode and other small edits

* Removed dashboard controls from playlist controls section and reformatted

* Added playlist controls behaviour in kiosk mode

* Updated Use dashboards image and descriptions

renovate[bot] committed:
Update dependency rc-cascader to v3.30.0 (#96444)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

samjewell committed:
Set CoreServices as codeowners for SQL Expressions (#96342)

* Set CoreServices as codeowners for SQL Expressions

* Set CoreServices as CodeOwners for ALL SSEs

Don't limit to SQL Expressions alone, but take all of the
Server Side Expressions

alexander-akhmetov committed:
Alerting: Add simplified_notifications_section field to the alert rule metadata (#95988)

ashharrison90 committed:
GrafanaUI: Mark `ScrollContainer` as stable and deprecate `CustomScrollbar` (#96331)

* deprecate CustomScrollbar, promote ScrollContainer

* add documentation start

* use the properly exported ScrollContainer

* Update packages/grafana-ui/src/components/ScrollContainer/ScrollContainer.mdx

Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>

---------

Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>

renovate[bot] committed:
Update dependency rc-tree to v5.10.1 (#96445)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

eledobleefe committed:
New Select: decrease item height (#95952)

renovate[bot] committed:
Update dependency react-calendar to v5.1.0 (#96449)

* Update dependency react-calendar to v5.1.0

* use properly exported CalendarType

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

andresmgot committed:
Preinstall: Allow to extend default list (#96402)

andresmgot committed:
Enable preinstallAutoUpdate by default (#96446)

renovate[bot] committed:
Update dependency react-dropzone to v14.3.5 (#96453)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Chore(deps): Bump http-proxy-middleware from 2.0.4 to 2.0.7 (#95682)

Bumps [http-proxy-middleware](https://github.com/chimurai/http-proxy-middleware) from 2.0.4 to 2.0.7.
- [Release notes](https://github.com/chimurai/http-proxy-middleware/releases)
- [Changelog](https://github.com/chimurai/http-proxy-middleware/blob/v2.0.7/CHANGELOG.md)
- [Commits](https://github.com/chimurai/http-proxy-middleware/compare/v2.0.4...v2.0.7)

---
updated-dependencies:
- dependency-name: http-proxy-middleware
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency rollup to v4.26.0 (#96456)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

grafana-pr-automation[bot] committed:
I18n: Download translations from Crowdin (#96440)

New Crowdin translations by GitHub Action

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

evictorero committed:
Docs: Add allow list email domain configuration (#95882)

renovate[bot] committed:
Update dependency sass to v1.80.7 (#96459)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

colin-stuart committed:
Auth: Passwordless Login Option Using Magic Links (#95436)

* initial passwordless client

* passwordless login page

* Working basic e2e flow

* Add todo comments

* Improve the passwordless login flow

* improved passwordless login, backend for passwordless signup

* add expiration to emails

* update email templates & render username & name fields on signup

* improve email templates

* change login page text while awaiting passwordless code

* fix merge conflicts

* use claims.TypeUser

* add initial passwordless tests

* better error messages

* simplified error name

* remove completed TODOs

* linting & minor test improvements & rename passwordless routes

* more linting fixes

* move code generation to its own func, use locationService to get query params

* fix ampersand in email templates & use passwordless api routes in LoginCtrl

* txt emails more closely match html email copy

* move passwordless auth behind experimental feature toggle

* fix PasswordlessLogin property failing typecheck

* make update-workspace

* user correct placeholder

* Update emails/templates/passwordless_verify_existing_user.txt

Co-authored-by: Dan Cech <dcech@grafana.com>

* Update emails/templates/passwordless_verify_existing_user.mjml

Co-authored-by: Dan Cech <dcech@grafana.com>

* Update emails/templates/passwordless_verify_new_user.txt

Co-authored-by: Dan Cech <dcech@grafana.com>

* Update emails/templates/passwordless_verify_new_user.txt

Co-authored-by: Dan Cech <dcech@grafana.com>

* Update emails/templates/passwordless_verify_new_user.mjml

Co-authored-by: Dan Cech <dcech@grafana.com>

* use &amp; in email templates

* Update emails/templates/passwordless_verify_existing_user.txt

Co-authored-by: Dan Cech <dcech@grafana.com>

* remove IP address validation

* struct for passwordless settings

* revert go.work.sum changes

* mock locationService.getSearch in failing test

---------

Co-authored-by: Mihaly Gyongyosi <mgyongyosi@users.noreply.github.com>
Co-authored-by: Dan Cech <dcech@grafana.com>

grafana-pr-automation[bot] committed:
I18n: Download translations from Crowdin (#96460)

New Crowdin translations by GitHub Action

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

JoaoSilvaGrafana committed:
Bookmarks: Lazy loading page (#96455)

JoaoSilvaGrafana committed:
New Select: Fix not being able to click around the caret (#96026)

renovate[bot] committed:
Update dependency @grafana/faro-core to v1.12.0 (#96461)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.12.0 (#96462)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/golang-jwt/jwt/v4 from 4.5.0 to 4.5.1 (#95807)

* Bump github.com/golang-jwt/jwt/v4 from 4.5.0 to 4.5.1

Bumps [github.com/golang-jwt/jwt/v4](https://github.com/golang-jwt/jwt) from 4.5.0 to 4.5.1.
- [Release notes](https://github.com/golang-jwt/jwt/releases)
- [Changelog](https://github.com/golang-jwt/jwt/blob/main/VERSION_HISTORY.md)
- [Commits](https://github.com/golang-jwt/jwt/compare/v4.5.0...v4.5.1)

---
updated-dependencies:
- dependency-name: github.com/golang-jwt/jwt/v4
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

* fix make workspace update

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Jo <me@jguer.space>

imatwawana committed:
Docs: Add canvas custom images and icon guidance (#91806)

Co-authored-by: Drew Slobodnjak <60050885+drew08t@users.noreply.github.com>

jtvdez committed:
Docs/adding information on using variable expansion and env variable for TLS certs when provisioning data sources (#96400)

Docs/adding information on using variable expansion and env variables for TLS certs when provisioning data sources

renovate[bot] committed:
Update scenes to v5.25.0 (#96463)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

kminehart committed:
Alerting: Add useReturnTo hook to safely handle returnTo parameter (#96474)

Add useReturnTo hook to safely handle returnTo parameter

Co-authored-by: Konrad Lalik <konrad.lalik@grafana.com>

aocenas committed:
Sidecar: Route handling rework (#96337)

renovate[bot] committed:
Update dependency @grafana/plugin-e2e to v1.12.1 (#96476)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

leeoniya committed:
XYChart: Remove old implementation (#96416)

IevaVasiljeva committed:
GAS: Document group attribute sync roles and permissions (#96087)

document GAS roles and permissions

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.12.0 (#96485)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @eslint/compat to v1.2.3 (#96487)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

JacobsonMT committed:
Alerting: Fix incorrect 500 code on missing alert rule dashboardUID / panelID (#96491)

owensmallwood committed:
Unified storage Indexer: Add kind to hardcoded resource types (#96149)

* add kind to hardcoded resource types for indexer

* call ListOptions in Watch

* Use GaugeVec instead of CounterVec

* index folders before dashboards

lucychen-grafana committed:
Reporting: Add theme option (#95008)

* swagger doc

* Update doc for report settings

* Apply suggestions from code review

* swagger doc

* add doc for current theme

* update text

* wording feedback changes

* make it clearer

* update again

* update theme component

* swagger doc

---------

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>

imatwawana committed:
Docs: Add private preview info for email sharing (#96350)

gelicia committed:
Codeowners: reflect changed ownership for explore squad (#96492)

juanicabanas committed:
Reporting: useReportRawTimeRange FF (#95683)

bohandley committed:
Combobox: Add a warning for combobox when exceeding the recommended amount of options (#96070)

* add a warning for components when they exceed the recommended options amount

* Update packages/grafana-ui/src/components/Combobox/Combobox.tsx

Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>

* Update packages/grafana-ui/src/components/Combobox/Combobox.tsx

Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>

* log faro

* log once with flag

* log in setItems

* move logOptions to utils and write tests

* import function, update betterer for combobox

* Fix lint errors

* log once without the try

* fix tests

---------

Co-authored-by: Tobias Skarhed <1438972+tskarhed@users.noreply.github.com>
Co-authored-by: joshhunt <josh@trtr.co>

edvard-falkskar committed:
NodeGraph: Zoom mode option (#95823)

* NodeGraph: Zoom mode option

* Update the docs

* refactor stepUp/stepDown handling

* fix

* Update cooperative description

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>

* Update zoom mode wording

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>

* Add  tests

* Move node graph settings to align with other viz

---------

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>

dmihai committed:
docs: Add docs on configuring role mapping for multiple GitHub teams (#96483)

add docs on configuring role mapping for multiple GitHub teams

vtorosyan committed:
Docs: Adding a note about how to parse JMESPath expression with a colon (#96237)

* Docs: Adding a note about how to parse JMESPath expression with a colon

---------

Co-authored-by: Irene Rodríguez <irene.rodriguez@grafana.com>

aocenas committed:
Grafana/ui: Allow adjusting the handle size of the useSplitter (#96039)

adrapereira committed:
Tempo: Fix trace kvlist and array transformations (#96388)

* Fix kvlist and array transformations

* Log transformation errors

joshhunt committed:
GrafanaUI: AutoSizeInput uses placeholder for width calculation (#96458)

AutoSizeInput: Use placeholder for width calculation also

renovate[bot] committed:
Update dependency @grafana/faro-web-tracing to v1.12.0 (#96493)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ashharrison90 committed:
SingleTopNav: Update `singleTopNav` feature toggle to public preview (#96527)

update singleTopNav to public preview

renovate[bot] committed:
Update dependency @floating-ui/react to v0.26.28 (#96528)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ashharrison90 committed:
Chore: silence sass warnings since we're going to remove sass anyway (#96530)

silence sass warnings since we're going to remove sass anyway

soniaAguilarPeiron committed:
Alerting: Remove reducer when creating and instant (#94246)

* Remove reducer when using a non complex query that is instant

* remove reducer when changing data source

* Fix whenField

* use DataSourceType instead of string literal

* add reducer when using range

* add tests

* use an object for SimpleCondition refids identifiers

* fix threshold expression to point to B after switching back to range

* address pr review comments

* refactor: extract reducer optimization to the reducer

* fix tests

* fix snapshot

* rename constants

kalleep committed:
Zanzana: Lazy load cached store info (#96452)

* Lazy load cached store infos

tolzhabayev committed:
Docs: Plugin Installation - reorganize and add air-gapped environments (#96437)

* rewriting installation docs and including info about airgapped environments

* Changing example to lokieexplore app

* run prettier

* adding corrections

* running prettier

---------

Co-authored-by: Irene Rodriguez <irene.rodriguez@grafana.com>

kalleep committed:
Zanzana: reconcile basic roles and bindings (#96473)

* Add reconciler for basic roles

* Add reconciler for basic role bindings

jackw committed:
SQL Datasources: Make QueryEditor lazy loaded (#96179)

* feat(grafana-sql): make sqlqueryeditor lazy loaded

* feat(sql-datasources): switch out query editor for lazy query editor

* build(postgresql): put tsconfig jsx prop inside compilerOptions

* chore(sql-queryeditor): move lazy import out of component

mdvictor committed:
DashboardScene: Update repeated panels in dashboard after saving in panel edit and returning to it (#96536)

* Update repeated panels after saving dashboard when returning to it from panel edit

* cleanup

svennergr committed:
Loki: Fix healthcheck logger always appending `endpoint` (#96531)

* Loki: Fix healthcheck logger always appending `endpoint`

* add test

codecapitano committed:
Frontend Monitoring: Update regex to ignore options.customEndpoint (#96542)

* fix(GrafanaJavascriptAgentBackend): update regex to ignore options.customEndpoint

* update test

vtorosyan committed:
Docs: Add secondsToLive to the SA HTTP API docs (#96518)

* Docs: Add secondsToLive param to the SA HTTP API docs

* Docs: Add secondsToLive param to the SA HTTP API docs

vtorosyan committed:
Docs: Clarify disabling auto login for SAML and OAuth (#96520)

tomratcliffe committed:
Chore: Move betterer eslint rules to separate file and allow opting in (#96240)

* Move betterer eslint rules to separate file and allow opt-in for local dev

* Add betterer eslint to codeowners

suntala committed:
K8s/Folders: Allow recursive creation of DTO (#96439)

* Fix toDTO
* Remove conversion function for folder dto
* Convert toDTO to a standalone function

---------

Co-authored-by: Jean-Philippe Quéméner <JohnnyQQQQ@users.noreply.github.com>

ryantxu committed:
Provisioning: Rename k8s origin metadata to repo (#96524)

Commit:
Jaeger: Add feature toggle for backend migration (#96494)

* add feature toggle for jaeger backend migration

* add codeowners

gtk-grafana committed:
Loki: skip tests that are flaking in CI (#96551)

test: skip tests that are flaking in CI

imatwawana committed:
Docs: Bar chart visualization refactor (#96555)

iwysiu committed:
Elasticsearch: Update grafana-plugin-sdk-go for sse errorsource updates (#96144)

dmihai committed:
docs: Add docs on generating SAML cert from the UI (#96538)

* add docs on generating SAML cert from the UI

* run prettier

JacobsonMT committed:
Alerting: Fix long KeepLast annotations failing to save (#96503)

Certain combinations of KeepLast state transition annotations exceed the current
 limit of 25 characters. This increases the column size limit to be able to
 handle the longest combination (27) with some extra space for comfort.

stephaniehingtgen committed:
Stars: Add dashboard_uid and org_id to table (#96408)

margotphelps committed:
Docs: Added New Release Dates,Security Versioning and Freezes (#96413)

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>
Co-authored-by: Isabel Matwawana <isabel.matwawana@grafana.com>

M7mdisk committed:
Chore: Update native histogram hints in prometheus data source (#96270)

* [Prometheus] Update native histogram hints

* update unit tests

colin-stuart committed:
Docs: add note that support for Auth0 "audience" feature is not available (#96507)

* add note that support for Auth0 audience feature is not available

* Update docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

---------

Co-authored-by: Misi <mgyongyosi@users.noreply.github.com>

yuri-tceretian committed:
Alerting: Update state manager to have immutable state in cache (#95985)

* create a new state and set at the end
* propagate labels datasource_uid and ref_id from current state if it's error
* copy the state when apply to all

iwysiu committed:
Cloudwatch: Fix parsing for ec2 resource attributes (#96501)

toddtreece committed:
Dashboards: Add v1alpha1 and v2alpha1 conversion (#96415)

---------

Co-authored-by: Stephanie Hingtgen <stephanie.hingtgen@grafana.com>

ryantxu committed:
Storage:  Revert using real gRPC for integration tests (#96410)


---------

Co-authored-by: Todd Treece <todd.treece@grafana.com>

charandas committed:
Cloud Aggregation: make it work for users with no name (#96509)

JacobsonMT committed:
Alerting: Fix k8s api route fingerprint to include ObjectMatchers and Provenance (#96591)

Fix route fingerprint to include ObjectMatchers and Provenance

konrad147 committed:
Alerting: New list view (Part 1) (#95039)

* initial commit

* update styles

* wip

* update list view

* update translations

* abstract components

* metadata separator

* refactor

* cleanup

* fix tests

* WIP

* translations

* refactor to use maps and type-safety

* WIP

* UI updates

* Rule action buttons early draft

* recording rules

* WIP typescript errors

* implement action button loading

* move section loader etc

* add placeholder for group actions

* Change files structure, remove CombinedRule from AlertRuleMenu

* Refactor fetching data sources with ruler

* Fix tests

* Unify data source features

* move files

* make actions column wider

* update translations

* Update tests to reflect code changes

* Remove direct buildinfo usages

* Fix useCanSilence hook

* Add missing translations, fix lint errors

* PR feedback

* update test

* Remove featureDiscovery mock from a test

---------

Co-authored-by: Gilles De Mey <gilles.de.mey@gmail.com>

ivanahuckova committed:
Zipkin: Move query running to backend (#96404)

* Rename method applyTemplateVariables so it used by backend

* Zipkin: Move query running to backend

* Fix error source

* Hndle invalid query and return error response

mgyongyosi committed:
Docs: Clarify MFA/2FA in the docs (#96610)

* Docs: Clarify MFA/2FA in the docs

* Update docs/sources/setup-grafana/configure-security/configure-authentication/_index.md

Co-authored-by: Vardan Torosyan <vardants@gmail.com>

---------

Co-authored-by: Vardan Torosyan <vardants@gmail.com>

IevaVasiljeva committed:
RBAC: Remove folder name scope resolver (#96484)

* remove folder name scope resolver

* update test

brendamuir committed:
Alerting docs: fix query syntax sentence (#96614)

* Alerting docs: fix query syntax sentence

* updates sentence

* pepes feedback

soniaAguilarPeiron committed:
Alerting: Add alertingOptimizeReducerInUI feature toggle (#96608)

add alertingOptimizeReducerInUI feature toggle

eledobleefe committed:
Combobox: Menu places itself under `topnav` (#96545)

samjewell committed:
SQL Expressions: Simplify where code is being stubbed out (#96227)

* Rename from DuckDB

* Remove unused function

svennergr committed:
Sidecar: Always hide at `/login` page (#96621)

aangelisc committed:
CloudMonitoring: Improve metric handling for `DISTRIBUTION` value types (#96412)

* Set crossSeriesReducer value based on metric valueType

* Update label retrieval to retrieve all values

* Update tests

* Update documentation

* Add tooltip

* fix typo

* Docs lint

---------

Co-authored-by: Irene Rodríguez <irene.rodriguez@grafana.com>

alexanderzobnin committed:
Zanzana: Search with check server side (#96268)

* pass zclient into dashboard service

* Search then check implementation

* Use GetNamespace() for user

* remove unused orgID

* simple batch check

* refactor

* add tests

* fix batchCheckItem

* client implements batch check

* use batch check in search

* remove unused

* remove All field from response

* refactor: extract checkNamespace

* fix search result uniqueness

* comment fix

* Apply suggestions from code review

Co-authored-by: Karl Persson <kalle.persson@grafana.com>

* refactor

* cleanup

* remove unnecessary check

* fix tests

* fix protobuf def

* Fix query page

* fix type

---------

Co-authored-by: Karl Persson <kalle.persson@grafana.com>

ivanahuckova committed:
Elasticsearch: Replace error source http client with a new error source methods (#96570)

* Elasticsearch: Replace error source http client with methods

* Update

* Update

* Update tests

* Update

* Update

* Update pkg/tsdb/elasticsearch/data_query.go

Co-authored-by: Nathan Vērzemnieks <njvrzm@gmail.com>

* Trigger build

---------

Co-authored-by: Nathan Vērzemnieks <njvrzm@gmail.com>

iwysiu committed:
AWS: Add amazon managed prometheus and aurora to the forward list (#96590)

AWS: add amazon managed prometheus and aurora to the forward list

imatwawana committed:
Docs: Logs visualization refactor (#96576)

bfmatei committed:
Dashboard: Fix dashboard reload behavior (#96427)

ashharrison90 committed:
UPlot: don't use state to hold reference to uplot instance (#96630)

don't use state to hold reference to uplot instance

yincongcyincong committed:
Explore Metrics: Duplicate subpath (#96287)

* explore metrics subpath

* optimize

NWRichmond committed:
Prometheus: Performant Code Mode autocomplete with long metric names (#96580)

* fix: autocomplete performance

* refactor: simplify the complex search strategy

* chore: allow json imports to aid testing

* perf: prefer for loop for perf-critical path

* perf: use ufuzzy in intraMode: 0 (#96584)

* refactor: add clarity

* refactor: simplify

---------

Co-authored-by: Leon Sorokin <leeoniya@gmail.com>

aocenas committed:
Sidecar: Add rudderstack events (#96566)

grafana-pr-automation[bot] committed:
I18n: Download translations from Crowdin (#96612)

New Crowdin translations by GitHub Action

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

JacobsonMT committed:
Alerting: Fix annotation state migration to retain nullability on MySQL (#96640)

owensmallwood commented on pkg/storage/unified/resource/search.go:
```diff
@@ -29,3 +31,54 @@ type DocumentBuilderInfo struct {
+		size int64,
+
+		// The last known resource version (can be used to know that nothing has changed)
+		resourceVersion int64,
```
The one thing I like about having an index per namespace + resource is it makes it easier if we want to back up the index. We can restore it and then Watch from it's rv to initialize it.

cedricziel committed:
Backend: Inject server-timing header to match initial loads with client-side telemetry (#94978)

feat: inject server-timing header to match initial loads with client-side telemetry

Faro Web SDK can read the server-timing info and
correlate the initial request with the client side telemetry gather from navigation timings.

yesoreyeram committed:
Datasources: Health check error tracking (#96629)

* datasource health check error tracking

* type fix

tonypowa committed:
Alerting-loki tutorial: update docker env (#96486)

zoltanbedi committed:
MySQL: Fix tls skip verify option (#96475)

MySQL: Fix skip tls verify option

linoman committed:
Auth: Add None role to Viewer case (#96624)

Add None role to Viewer case

gillesdemey committed:
Alerting: Use NestedFolderPicker (#96398)

Co-authored-by: Tom Ratcliffe <tom.ratcliffe@grafana.com>

renovate[bot] committed:
Update dependency react-dropzone to v14.3.5 (#96454)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

volcanonoodle committed:
Update JWT docs (#96623)

tskarhed committed:
Combobox: Support undefined, null value and improve typing (#96523)

* Support undefined value

* Check truthiness of value instead

* check falsy

* Conditional typing for clearing value

* Less restrictive default typing

* simplify props

* Add tests for autosizing

* Write failing test case

* Add list of falsy values

* Check if nullish

* Check nullish in itemToString

* Nvm, it doesn't matter here

* Add support for autoFocus

* Pick from InputProps

* Move docstring

* Solve type issues in Storybook

* Fix failing story

gillesdemey committed:
Alerting: Silence drawer being forcefully closed (#96579)

Co-authored-by: Tom Ratcliffe <tom.ratcliffe@grafana.com>

imatwawana committed:
Docs: Node graph visualization refactor (#96647)

axelavargas committed:
Transform DashboardScene into Schema V2 (#95546)

---------

Co-authored-by: Dominik Prokop <dominik.prokop@grafana.com>
Co-authored-by: Ivan Ortega Alba <ivanortegaalba@gmail.com>
Co-authored-by: Haris Rozajac <haris.rozajac12@gmail.com>

kalleep committed:
Zanzana: sync roles and bindings (#96661)

* Sync fix and custom roles and team bindings.

* Add collector for user role bindings

stephaniehingtgen committed:
K8s: dashboards: fix conversion (#96648)

chaudyg committed:
unistore: wire the authz client (#96632)

* unistore: wire the authz client

* rename dashboards.grafana.app into dashboard.grafana.app

* wire the authz client

* wire the authz client

* resuse the Standalone constructor

* configure default migration for resource folder

* add tests

* cleanup

* add logging

kalleep committed:
Zanzana: Optimize batch check (#96669)

* Restructure check code so we only check namespace access once for each GroupResource during for batch

ryantxu committed:
UnifiedStorage: Add ResourceTable format  (#96506)

renovate[bot] committed:
Update dependency @formatjs/intl-durationformat to v0.6.4 (#96662)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

joshhunt committed:
GrafanaUI: Size AutoSizeInput correctly when used with suffix/prefix (#96575)

* GrafanaUI: Size AutoSizeInput correctly when used with suffix/prefix

* comment

* reword comment

gelicia committed:
Correlation: Fix loki dataplane field change (#94088)

* Fix loki dataplane field change

* Add test for dataplane

* Reset feature toggle in test

ashharrison90 committed:
Internationalisation: Add basic e2e test to ensure languages load correctly (#96556)

* fix pseudo locale and add simple e2e test

* add comment

* update test and add comments

* fix unit test

* try 5 instead of 6?

* clear input

* only check pseudo in dev mode

* don't check pseudo in CI

* don't test pseudo locale

* isolate language changes

* move waits around

* only wait for preferences on first visit

* add proper loading state

* bit of tidy up

* hook up ids correctly

* use disabled/isLoading instead of skeleton

* set props on Field

renovate[bot] committed:
Update dependency @grafana/faro-core to v1.12.1 (#96684)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

matyax committed:
Explore: Sync logs displayed fields with URL state (#96242)

* Explore: store displayedFields in URL state

* Displayed fields: reset when queries change

* Add test

* LogRowMessageDisplayedFields: pass logRowMenuIcons props to component

* LogsMetaRow: use primary outline for reset button

* Logs: clear displayedFields from URL

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.12.1 (#96685)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Commit:
Release: update changelog for 10.4.13 (#96674)

Update changelog

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.12.1 (#96688)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

adrapereira committed:
Tempo: Run TraceQL metrics queries through backend (#96246)

* Move TraceQL metrics queries to backend

* Add tests for TransformMetricsResponse

* Add tests for createMetricsQuery

* Refactor, detect metric queries and throw error for search queries

* Remove unneeded types

* Don't add start and end if they are 0. Return on err

* lint

* Fix pre-alloc of frames

* Address PR comments

svennergr committed:
Logs: Let `level` label take precedence over `detected_level` (#96678)

gelicia committed:
Correlations: Handle special field for loki labels (using data plane) (#94033)

Use dataplane data if available

Commit:
Release: update changelog for 11.0.8 (#96690)

Update changelog

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

soniaAguilarPeiron committed:
Alerting: Move evaluation outside folder section, and move labels in instead (#95121)

* Move evaluation outside folder section, and move labels in instead

* rename file

* update translations

* refactor

* rename file and component

* refactor

* fix test

* refactor

* rename files and components

* update translations

* fix style

* update translations

* Use useAppNotification for toasts

* update label when group can not be selected yet

* update translations

* update some texts and add comment

* update translations

* remove duplicated code

* fix typo

* update texts and translations

* rename FolderWithoutGroup to FolderSelector

* restore wrong updates

* restore wrong updates

* translations and remove GroupAndFolder component

* address review comments

* remove container

* address review comments

* address review comments

* prettier

* prettier

renovate[bot] committed:
Update dependency @grafana/faro-web-tracing to v1.12.1 (#96698)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Commit:
Release: update changelog for 11.1.9 (#96692)

Update changelog

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Andreas Christou <andreas.christou@grafana.com>

Commit:
Release: update changelog for 11.2.4 (#96691)

Update changelog

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Andreas Christou <andreas.christou@grafana.com>

renovate[bot] committed:
Update dependency eslint-plugin-unicorn to v56.0.1 (#96709)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

matyax committed:
Logs: Allow scroll to reach the bottom of the log list before loading more (#96668)

* Infinite scroll: use timestamps to improve scrolling experience

* Add unit tests

baldm0mma committed:
CI: Fix `latest` tag not being correctly applied to releases (#96497)

* baldm0mma/tag_latest_bug/ update github-release.yml default values

* baldm0mma/tag_latest_bug/ update release-comms.yml

* baldm0mma/tag_latest_bug/ update logic

* baldm0mma/tag_latest_bug/ update logic

* baldm0mma/tag_latest_bug/ update logic

* baldm0mma/tag_latest_bug/ update logic for both cases

* baldm0mma/tag_latest_bug/ update comment

renovate[bot] committed:
Update dependency i18next to v23.16.6 (#96715)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

Commit:
Release: update changelog for 11.3.1 (#96713)

Update changelog

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>
Co-authored-by: Andreas Christou <andreas.christou@grafana.com>

IevaVasiljeva committed:
GAS: update auth provider docs (#96097)

* add docs describing what GAS is and how it can be configured

* auth provider docs for GAS

* PR feedback

kaydelaney committed:
PanelInspectDrawer: Fix issue where closing inspect drawer on home dashboard redirects to new dashboard (#96711)

svennergr committed:
Loki: Allow regex in `label` derived field (#96609)

* Loki: Allow regex in `label` derived field

* add docs

* change to caution

leeoniya committed:
Transformations: Add RegExp option to Extract fields transformer (#96593)

margotphelps committed:
Added docket tag note to security section (#96725)

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>

simPod committed:
Docker: Don't use legacy ENV syntax (#93218)

chore(docker): don't use legacy ENV syntax

bohandley committed:
Prometheus: Add deprecation message for AMP in Promehteus docs (#96689)

deprecate message for AMP in prom docs

bohandley committed:
Explore metrics: Default for OTel to select deployment environment like prod (#96585)

* default to select deployment environment like prod

* add tests

* return null if options are empty to prevent error, fix test

oscarkilhed committed:
Scenes: Fix angular migrations that are using the targets property on the old panel model (#96626)

Fix angular migrations using target

jdbaldry committed:
Update Explore CODEOWNERS (#93531)

renovate[bot] committed:
Update dependency knip to v5.37.1 (#96718)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

gamab committed:
AuthZ: Launch service within IAM app (#96421)

renovate[bot] committed:
Update dependency @leeoniya/ufuzzy to v1.0.16 (#96743)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

grafana-pr-automation[bot] committed:
I18n: Download translations from Crowdin (#96671)

New Crowdin translations by GitHub Action

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

yesoreyeram committed:
Datasources: health check logging improvements (#96403)

* basic healthcheck logging

* show detailed error for admins

* fix linting errors

* fix config summary keys

* passing context to logger

* tracking health check errors

* remove tracking

eledobleefe committed:
SingleTopNav: Update `singleTopNav` feature toggle to GA and enabled by default (#96752)

dmihai committed:
Optimize tokens count from service accounts query (#96663)

* optimize tokens count from service accounts query

* add unit tests for tokens count

* skip broken test

* fix lint error

* rename Tokens to TokenCount

ivanortegaalba committed:
Add tests to SceneToSaveModelV2 transformer (#96700)

* Add tests to SceneToSaveModelV2 transformer

* Add text properties always

ashharrison90 committed:
Chore: Enhance debug theme with more hover colors (#96753)

enhance debug theme with more hover colors

torkelo committed:
Dashboard: Panel edit and support for different layout items (#96203)

* Dashboard: Panel edit and support for more layout items

* It's working

* Fix discard issue

* remove unused file

* Update

* Editing for responsive grid items now work

* Update

* Update

* Review fix

ryantxu committed:
Swagger: show user icon (#96736)

renovate[bot] committed:
Update dependency msw to v2.6.5 (#96745)

* Update dependency msw to v2.6.5

* add TransformStream to jest-setup

---------

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Ashley Harrison <ashley.harrison@grafana.com>

eleijonmarck committed:
Serviceaccounts: FIX adds qoute for user UID query for postgres (#96749)

* Add qoute for user UID of postgres

* add tests for uid

* fix for uid check not expectable

juanicabanas committed:
PublicDashboards: Remove publicDashboards FF (#96578)

renovate[bot] committed:
Update dependency dompurify to v3.2.1 (#96762)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

asimpson committed:
ds-querier: account for -100 meaning __expr__ (#96565)

* ds-querier: account for -100 meaning __expr__

* temp debug log error parsing query

* var confusion

* log error

renovate[bot] committed:
Update dependency i18next to v23.16.8 (#96765)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

papagian committed:
Update alpine version in Dockerfile (#95051)

Update Dockerfile

renovate[bot] committed:
Update dependency rudder-sdk-js to v2.48.29 (#96768)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

ivanahuckova committed:
Fix error source for `clientSecret not found` error (#96771)

* Fix error source for clientSecret not found errir

* Use errors.New as we are not formatting the error

renovate[bot] committed:
Update dependency @grafana/experimental to v2.1.4 (#96777)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

matyax committed:
Loki: Added support for disabled operations in Query Builder (#96751)

* Loki operations: define basic toggleable operations

* LokiQueryModeller: skip disabled operations

* Loki operations: define toggleable operations

* Add test

* Add test

renovate[bot] committed:
Update emotion monorepo (#96780)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

renovate[bot] committed:
Update scenes to v5.25.1 (#96782)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

lwandz13 committed:
Docs:  Added datasources codeowner (#96784)

added lwandz13 as datasources codeowner

IevaVasiljeva committed:
GAS: Show mapped roles in role picker (#96681)

* add group mapping UID returned mapped roles

* request mapped roles from the frontend, but don't attempt to update mapped roles

* lock mapped roles and show a pop-up message about why a role is locked

* update role selectors to not allow deselecting a mapped role

* swagger gen

* simplify and set mapped as bool instead of mapping UID array

* swagger gen

renovate[bot] committed:
Update dependency @grafana/faro-core to v1.12.2 (#96786)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

evictorero committed:
Docs: Add domain allow list report docs (#96778)

* Docs: Add domain allow list report docs

* Update docs/sources/setup-grafana/configure-grafana/enterprise-configuration/index.md

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>

---------

Co-authored-by: Isabel Matwawana <76437239+imatwawana@users.noreply.github.com>

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.12.2 (#96788)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

torkelo committed:
PanelEditor: Tweak category design (#96760)

* PanelEditor: Tweak category design

* Update

* remove unused import

* Switch angle dirctions

* Fix test

renovate[bot] committed:
Update dependency @grafana/faro-web-sdk to v1.12.2 (#96789)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>

imatwawana committed:
Docs: Moved and renamed navigation section (#96722)

imatwawana committed:
Docs: stat visualization refactor (#96729)

jtvdez committed:
Docs: Removing outdated versioning info in create reports (#96804)

Removing outdated versioning info in create reports

scottlepp committed:
fix title sort (#96807)

fix sort

svennergr committed:
Explore Metrics: Add button to add to exploration (#95637)

* Explore Metrics: Add button to add to exploration

* feat: enable explorations for breakdown panels

* refactor: types

* refactor: code should reflect single expected link

* test: exploration button basics

* refactor: leverage link limit

* feat: add exploration button to more panels

* chore: update betterer results

* test: refactor for clarity

* test: prefer consistent approach

---------

Co-authored-by: Nick Richmond <nick.richmond@grafana.com>

ryantxu committed:
UnifiedSearch: Introduce a DocumentBuilder interface (#96738)

kalleep committed:
Zanzana: Add setting zanzana reconciliation interval (#96687)

Add config option under RBAC for zanzana reconciliation interval

soniaAguilarPeiron committed:
Alerting: Fix bug when saving a rule more than once (#96658)

* Update existing property once we save the new rule data and stay in the same page

* prettier

* use cache invalidation for updating the existing property

* fix cloud usecase

gamab committed:
Fix: Correct handling of base64 padding during `aes-gcm` private key decryption.  (#96761)

* SecretsService: Use RawStdEncoding to avoid padding

* Commment

* Forgot one line

* Backward compatibility

dastrobu committed:
[docs] document 409 Conflict on updating data sources. (#93170)

tonypowa committed:
Tonypowa/tutorials (#96747)

* replaced request bin

* advanced options

* test data + screenshot

kalleep committed:
Zanzana: Capabilities api (#96682)

* Add capabilities api

bfmatei committed:
Scopes: Refactor the way scope paths are created (#96712)

chaudyg committed:
unistore: handle auth when fallback is used (#96772)

* handle auth when fallback is used

* handle auth when fallback is used

* add traces

tskarhed committed:
Angular: Remove unused admin pages (#96526)

* Remove unused Angular admin pages

* Update betterer

aocenas committed:
Sidecar: Specify where it can be opened and when it automatically closes (#96683)

joshhunt committed:
Chore: Make ResizeObserver test mock only emit if an element is being observed (#96770)

* Chore: Make ResizeObserver test mock only emit if an element is being observed

* update test that relied on old mock behaviour

ryantxu committed:
merge main
